contrib/devtools/README.md

```diff
@@ -2,6 +2,47 @@ Contents
 ========
 This directory contains tools for developers working on this repository.
 
+archive-binaries.py
+===============
+
+Script to copy all the binaries (bitcoind, bitcoin-tx, bitcoin-cli,
+qt/bitcoin-qt) to a bin and bin/qt directories (assumed to exist in
+the calling directory), appending the last commit id (and optionally
+an arbitrary string before the commit id passed ar parameter before
+the commit id) to the file name. 
+
+This way one can keep a collection of binaries for many different
+branches for testing (while clearly ditinguishing which is which).
+
+Example usage:
+
+```
+git fetch origin
+git checkout origin/v0.12.0
+make check -j21
+python contrib/devtools/archive-binaries.py v0.12.0
+./bin/bitcoind-v0.12.0-188ca9c
+
+git checkout origin/master
+git clean -fdx -e bin/
```

Comment:
> I can guarantee I didn't discovered -fdx by myself.

Hah! Using `git clean -fdx` is kind of black magic, you should first sign a contract in blood before you're allowed to use it. Don't ever recommend anyoen to use it without strong warning. Some may store ancillary development scripts, even wallets in the git tree.
Personally I have it aliased to `git napalm`.

> Oh, sorry, actually what we want is to explicitly include files for cleaning, not exclude them. Is there an option for that?

I don't know, I think it could be useful in some cases to add some extra files to delete. On the other hand, I build system changes happen so fast, I somehow doubt whether anyone is going to maintain a list of old files (unless you're volunteering).